### PR TITLE
[DGS-5745, CIAM-2563] Add Stream Catalog roles to CLI

### PR DESCRIPTION
**Pull request by lucy-fan**
Checklist
---------
1. [CRUCIAL] Is the change for CP or CCloud functionalities that are already live in prod?
   * yes: ok

What
----
DataSteward and DataDiscovery roles are planning to go GA this month, so this PR updates the CLI and adds a feature flag for these roles and some unit tests to make sure the roles are hidden behind the feature flag.

References
----------
https://confluentinc.atlassian.net/browse/DGS-5745

Test & Review
-------------
Tested in stag (create/delete DataSteward and DataDiscovery, list roles with and without flag to make sure DS/DD are present and not present respectively).

### CODE REVIEW

**Comment by lucy-fan** [comment]
Just realized that the `describe` also had to be updated with the new namespaces, so I've done that as well. @brianstrauch @MuweiHe please review when you get the chance.

Stag testing:
```
❯ ./confluent iam rbac role describe DataSteward  
+---------------+------------------------------------------------------------------------+
| Name          | DataSteward                                                            |
| Access Policy | [                                                                      |
|               |   {                                                                    |
|               |     "bindingScope": "environment",                                     |
|               |     "bindWithResource": false,                                         |
|               |     "allowedOperations": [                                             |
|               |       {                                                                |
|               |         "resourceType": "Topic",                                       |
|               |         "operations": ["Describe", "ReadCatalog", "WriteCatalog"]      |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "StreamLineage",                               |
|               |         "operations": ["View"]                                         |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "CatalogTagDefinition",                        |
|               |         "operations": ["Read", "Write", "Delete"]                      |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "CatalogBusinessMetadataDefinition",           |
|               |         "operations": ["Read", "Write", "Delete"]                      |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "Cluster",                                     |
|               |         "operations": ["AccessMetrics", "View", "AccessWithToken"]     |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "OwnSrClusterApiKey",                          |
|               |         "operations": ["Describe", "Alter", "Delete", "Create"]        |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "Pipeline",                                    |
|               |         "operations": ["Describe"]                                     |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "Subject",                                     |
|               |         "operations": [                                                |
|               |           "Delete",                                                    |
|               |           "Read",                                                      |
|               |           "Write",                                                     |
|               |           "ReadCompatibility",                                         |
|               |           "ReadCatalog",                                               |
|               |           "WriteCompatibility",                                        |
|               |           "WriteCatalog"                                               |
|               |         ]                                                              |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "SchemaRegistry",                              |
|               |         "operations": ["Describe", "AccessMetrics", "AccessWithToken"] |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "CloudCluster",                                |
|               |         "operations": ["Describe"]                                     |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "Connector",                                   |
|               |         "operations": ["Describe", "AccessMetrics", "AccessWithToken"] |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "KsqlCluster",                                 |
|               |         "operations": ["Describe", "AccessMetrics", "AccessWithToken"] |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "Environment",                                 |
|               |         "operations": ["Describe"]                                     |
|               |       }                                                                |
|               |     ]                                                                  |
|               |   },                                                                   |
|               |   {                                                                    |
|               |     "bindingScope": "organization",                                    |
|               |     "bindWithResource": false,                                         |
|               |     "allowedOperations": [                                             |
|               |       {                                                                |
|               |         "resourceType": "NotificationSubscription",                    |
|               |         "operations": ["Describe"]                                     |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "NotificationIntegration",                     |
|               |         "operations": ["Describe"]                                     |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "Organization",                                |
|               |         "operations": ["Describe"]                                     |
|               |       },                                                               |
|               |       {                                                                |
|               |         "resourceType": "SupportPlan",                                 |
|               |         "operations": ["Describe"]                                     |
|               |       }                                                                |
|               |     ]                                                                  |
|               |   }                                                                    |
|               | ]                                                                      |
|               |                                                                        |
+---------------+------------------------------------------------------------------------+

```

```
❯ ./confluent iam rbac role describe FakeRole   
Error: unknown role "FakeRole"

Suggestions:
    The available roles are: CloudClusterAdmin, DataDiscovery, DataSteward, DeveloperManage, DeveloperRead, DeveloperWrite, EnvironmentAdmin, KsqlAdmin, MetricsViewer, Operator, OrganizationAdmin, ResourceOwner.
```

File: internal/cmd/iam/command_rbac_role_describe.go [hunk 1]
```diff
@@ -40,30 +41,30 @@ func (c *roleCommand) describe(cmd *cobra.Command, args []string) error {
...
-				opts := &mdsv2alpha1.RolenamesOpts{Namespace: dataplaneNamespace}
-				dataplaneRolenames, _, _ := c.MDSv2Client.RBACRoleDefinitionsApi.Rolenames(c.createContext(), opts)
-				rolenames := append(publicRolenames, dataplaneRolenames...)
-
-				suggestionsMsg := fmt.Sprintf(errors.UnknownRoleSuggestions, strings.Join(rolenames, ", "))
-				return errors.NewErrorWithSuggestions(fmt.Sprintf(errors.UnknownRoleErrorMsg, role), suggestionsMsg)
-			}
 
+	if r.StatusCode == http.StatusNotFound {
+		roleNameOpts := &mdsv2alpha1.RolenamesOpts{Namespace: namespaces}
```
Comment (brianstrauch) [inline internal/cmd/iam/command_rbac_role_describe.go:60]:
```suggestion
		opts := &mdsv2alpha1.RolenamesOpts{Namespace: namespaces}
```

**Review by brianstrauch — APPROVED** [review]

**Comment by lucy-fan** [comment]
@brianstrauch I seem to be getting a 404 when I try to look at the semaphore job. I see that it's at `https://confluent-cli.semaphoreci.com/`, and I can access the confluent semaphore just fine, do I need to login somewhere or get special permissions to see it?

**Comment by brianstrauch** [comment]
@lucy-fan Are you logged in to Semaphore with GitHub? It should be publicly accessible to anyone.

**Comment by lucy-fan** [comment]
> @lucy-fan Are you logged in to Semaphore with GitHub? It should be publicly accessible to anyone.

I am logged into Semaphore and my account is connected to Github, if that's what you mean. :( I had some of my teammates try as well and they were also getting a 404. We might need to bring it up to dev-prod

**Comment by brianstrauch** [comment]
@elismaga can you help out here? Any reason why only the CLI team would have access to the new Semaphore organization?

**Review by pankydcoder — APPROVED** [review]
I have tested the commands and the results are documented [here](https://confluentinc.atlassian.net/wiki/spaces/DG/pages/2992446443/Stream+Catalog+RBAC+CLI+testing). LGTM!

**Comment by brianstrauch** [comment]
@lucy-fan FYI: While we wait for you to get access, Semaphore is showing an error with `make lint`.

**Pull request merged**